Adding action/epics addTransformer feature to print module

jdugh opened the pull request:
Same PR as this one : https://github.com/geosolutions-it/MapStore2/pull/9309
Include migration reducer to epics.

## Description

Adding addTransformer feature to the print module reducer/action for use this function outside of the MapStore2 framework code (like an MapStore2-geOrchestra extension).

<!-- A few sentences describing the overall goals of the pull request' s commits. -->

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [X] Feature

<!-- add here the ReadTheDocs link (if needed) -->

## Issue

**What is the current behavior?**
When you develop a new extension for Mapstore2-Georchestra, if you [import addTransformer](https://github.com/geosolutions-it/MapStore2/blob/a00d669c5149d8546bb8fcd6a853929abdd672a4/web/client/plugins/Print.jsx#L194-L201) in your MapStore2-geOrchestra extension code and use it, the adding tranformer is in a local copy of the final userTransformerChain variable.
[For now, components retrieved from MapStore (using the import) will be a copy of the existing ones](https://github.com/geosolutions-it/MapStoreExtension#limitations)

So, when you open the print module and click on "Print", the print module call [getSpecTransformerChain](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/utils/PrintUtils.js#L310) and use the variable userTransformerChain  which is not the same as your local copy userTransformerChain  modify by your local copy of addtransformer.
Your transform is not apply.


#<issue>

**What is the new behavior?**
For fix it and use addTransform from outside (MapStore2-geOrchestra extension), I propose to add an action and reducers to the print module for calling the addTransform methode through an action ; Print module in catch this action and apply the addTransformer method in the framework.

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications -->

## Other useful information

jdugh committed:
Adding action/epics addTransformer feature to print module

tkohr reviewed (approved):
Thanks for the backport @jdugh !

jdugh committed:
Print Epics : Observable.of to Observable.empty 

The previous Rx.Observable.of(addTransformer [...] make a bug.
addTransformer return no one (action or anything), and all code are synch.
So, call addTransformer and return an empty Observable.

jdugh committed:
Fix addTransformer epics by returning à new action 'PRINT_TRANSFORMER_ADDED' instead of a null action; fix unit test in this sense
